core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -91,6 +91,50 @@ private[spark] object ClosureCleaner extends Logging {
     (seen - obj.getClass).toList
   }
 
+  /** Initializes the accessed fields for outer classes and their super classes. */
+  private def initAccessedFields(
+      accessedFields: Map[Class[_], Set[String]],
+      outerClasses: Seq[Class[_]]): Unit = {
```

Comment:
what if multiple outer classes have the same parent class?

Reply:
I think from the view of closure, even multiple outer classes have the same parent class, the access of the fields in the parent class shouldn't conflict.

Reply:
I added a related test. Please see if it can clarify your concern.